src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -281,19 +281,14 @@ class Tensor<X, Symm, IndexList<Indices...>> {
                   "The index types (SpatialIndex or SpacetimeIndex) of the "
                   "generic indices in the expression do not match the index "
                   "types of the indices in the Tensor.");
-    if constexpr (tmpl::is_set<tmpl::size_t<TensorIndices::value>...>::value) {
-      return TE<tmpl::list<TensorIndices...>>(*this);
-    } else {
-      using rhs_tensorindex_list = tmpl::list<TensorIndices...>;
-      using rhs_tensorindex_values =
-          tmpl::list<tmpl::size_t<TensorIndices::value>...>;
-      using repeated_rhs_tensorindex_values = repeated<rhs_tensorindex_values>;
-      return TensorExpressions::detail::fully_contract<
-          TE,
-          replace_indices<rhs_tensorindex_list,
-                          repeated_rhs_tensorindex_values>,
-          1, tmpl::size<repeated_rhs_tensorindex_values>::value>(*this);
-    }
+    using tensorindex_value_list =
```

Comment:
Why can't the logic here for `placed_tensorindex_value_list`, etc. be inside contract? I was initially hoping this function would just be:
```cpp
return contract(TE<tmpl::list<TensorIndices...>>{*this});
```

Reply:
Nuked in 062e2b3